### DISCUSSION TITLE
자바에서 쓰이는 final 키워드에 대해서 설명해주세요

### DISCUSSION THREAD

**Discussion by kunsanglee** [2024-woowacourse-study/level-interview#60] [Java]
.

**Comment by J-I-H-O** [comment]
자바에서의 final 키워드는 변수, 메서드, 클래스에 사용할 수 있습니다.

변수에 대한 final 키워드의 경우, 값이 한 번만 할당될 수 있음을 의미합니다. 즉, 한 번 초기화된 후에는 값을 변경할 수 없습니다.
단 primitive type이 아닌 객체의 경우, 완전한 불변을 보장하는 것은 아닙니다. (꼬리질문)

메서드에 대한 final 키워드의 경우, 메서드가 하위 클래스에서 오버라이드 될 수 없음을 의미합니다.

클래스에 대한 final 키워드의 경우, 해당 클래스가 다른 클래스에 의해 상속될 수 없음을 의미합니다.

**Comment by 3Juhwan** [comment]
자바의 final 키워드는 변수, 메소드, 클래스에 사용합니다. 변수에 사용하면 값을 변경할 수 없게 만들고, 메소드에 사용하면 하위 클래스에서 재정의할 수 없게 하며, 클래스에 사용하면 상속을 금지합니다. 이를 통해 안정성을 높이고 코드의 의도를 명확히 전달할 수 있습니다.
